core/domain/draft_upgrade_services.py

```diff
@@ -88,6 +88,36 @@ def try_upgrading_draft_to_exp_version(
 class DraftUpgradeUtil(python_utils.OBJECT):
     """Wrapper class that contains util functions to upgrade drafts."""
 
+    @classmethod
+    def _convert_states_v32_dict_to_v33_dict(cls, draft_change_list):
+        """Converts draft change list from state version 32 to 33. State
+        version 33 adds showChoicesInShuffledOrder boolean variable to the
+        MultipleChoiceInput interaction.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+        for i, change in enumerate(draft_change_list):
+            if (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                    change.property_name ==
+                    exp_domain.STATE_PROPERTY_INTERACTION_CUST_ARGS):
+                if change.new_value.keys() == ['choices']:
```

Comment:
We need to check for a specific interaction(multiple choice input) because the ```showChoicesInShuffledOrder``` should only be included in multipleChoiceInput and not in any other interaction. Does this answer your question?

Reply:
So why we haven't checked that for which interaction cut arg we are populating this `showChoicesInShuffledOrder` value?

Reply:
The cust arg schema is specifically for multipleChoiceInput interaction. I tried to check for the the interaction type and after sean's feedback I changed it to this format.

Reply:
Have you tried with ItemSelectionInput interaction? I see `choices` as a key for cust_arg in that interaction as well.
![image](https://user-images.githubusercontent.com/16653571/79363416-7cc60c80-7f65-11ea-8693-d0e207167a0c.png)

Reply:
@DubeySandeep I have accounted for itemSelectionInput and dragAndDropSortInput which has choices in cust arg. But they have other args which makes it differ from multipleChoiceInput. I have also included itemSelectionInput in the tests to verify that it doesnot create problems.

Reply:
It won't create a problem right now but these (unwanted) data will be saved in the exp models, right? If yes, then this can create an unexpected problem in the future, right?

Reply:
Can you show me the interaction data after we apply the upgraded change_list having change cmd for cust arg of item selection interaction?

Reply:
I am a little bit confused as to what I am supposed to do. As for your last question I have added a test that contains cust arg for item selection interaction, is that what you are asking for?

Reply:
Can you test this manually:
1. Start the server (on develop branch) and create 1 exploration.
2. Add ItemSelectionInteraction (in draft)
3. Switch to (shuffle-multiple-choice) feature branch.
4. Refresh the editor page.
5. Save the draft.
6. Log the exploration data (mostly the interaction part of the state having ItemSelection interaction.)

Add the log here.

(You can try this using backend services as well.)

Reply:
I am sorry I didn't realize you were asking me to do migration tests with itemSelectionInput. I have already tested it out. I have attached the interaction data alone that was sent to the backend when saving the draft in shuffle-multipe-choice branch. The draft was not affected.
[draft_data.txt](https://github.com/oppia/oppia/files/4496694/draft.txt)

Reply:
Thanks for checking. I realised the tricky part here (I was thinking  it's `if 'choices' in change.new_value.keys()` but it was `if change.new_value.keys() == ['choices']`) Sory for the trouble and thanks for the explanation.